tests/unit/test_utils.py

```diff
@@ -57,3 +59,15 @@ def test_exclude_variants_excludes_pages_with_metadata_not_canonical():
     page.personalisation_metadata.is_canonical = False
     result = exclude_variants([page])
     assert result == []
+
+
+def test_exclude_variants_with_pages_querysets():
+    '''
+    Test that excludes variant works for querysets too
+    '''
+    page1 = WagtailPage(path="/", depth=0, url_path="/", title="Hoi")
+    page2 = WagtailPage(path="/", depth=0, url_path="/", title="Hoi")
+    pages = WagtailPage.objects.all().order_by('id')
+    print(pages)
+    result = exclude_variants(pages)
+    assert result == pages
```

Comment:
Please add a page that `excludes_variants` wouldn't keep in the list so that this is testing that still works.
Also you can test that the classes of `result` and `pages` match.